lib/new_relic/tracer/report.ex

```diff
@@ -67,67 +67,37 @@ defmodule NewRelic.Tracer.Report do
   end
 
   def call(
-        {module, function, args},
+        {module, function, _args} = mfa,
         {name, category: :external},
         pid,
-        {id, parent_id},
-        {start_time, start_time_mono, end_time_mono}
+        ids,
+        times
       ) do
-    arity = length(args)
-
-    maybe_url =
-      case args do
-        [first | [second | _]] when is_atom(first) and is_binary(second) -> second
-        [first | _] when is_binary(first) -> first
-        _ -> nil
-      end
-
-    {long_name, short_name} =
-      if maybe_url && String.match?(maybe_url, ~r/\Ahttp/) do
-        target_host_name = URI.parse(maybe_url) |> Map.get(:host)
-        {target_host_name, target_host_name}
-      else
-        {
-          function_name({module, function, arity}, name),
-          function_name({module, function}, name)
-        }
-      end
+    metric_name = function_name({module, function}, name)
 
-    __MODULE__.call(
-      {module, function, args},
-      {name, category: :external, reported_name: {short_name, long_name}},
-      pid,
-      {id, parent_id},
-      {start_time, start_time_mono, end_time_mono}
-    )
+    __MODULE__.call(mfa, {name, category: :external, metric_name: metric_name}, pid, ids, times)
```

Comment:
I think it will be safer if we keep a single `call` signature, then pattern match this name tuple internally. That way we avoid unwanted recursions. Apart from that, this whole thing looks pretty good 👌.

Reply:
Another idea to consider: replacing this keyword list with a map, e.g. `%{name: name, category: :external, metric_name: metric_name}`. It's a definitely a bit more verbose, also a breaking change in the API, which would require careful deprecation, but IMO it paves the way for better extensibility.